From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Don Dutile <ddutile@redhat.com>
Date: Wed, 19 Feb 2020 11:52:18 +0100
Subject: [PATCH] redhat: rh_kabi: Indirect EXTEND macros so nesting of other
 macros will resolve.

RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: Jonathan Toppins <jtoppins@redhat.com>
RH-Acked-by: Jiri Benc <jbenc@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

Upstream status: RHEL only

rh_kabi: Indirect EXTEND macros so nesting of other macros will resolve.

Upstream Status: RHEL only
Suggested-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Donald Dutile <ddutile@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
---
 include/linux/rh_kabi.h | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/include/linux/rh_kabi.h b/include/linux/rh_kabi.h
index 9a4fea93ad99..cdc636d3013d 100644
--- a/include/linux/rh_kabi.h
+++ b/include/linux/rh_kabi.h
@@ -210,13 +210,17 @@
  * a pointer in a base struct.  The name of the new struct is the name
  * of the base struct appended with _rh.
  */
-#define RH_KABI_SIZE_AND_EXTEND_PTR(_struct)				\
+#define _RH_KABI_SIZE_AND_EXTEND_PTR(_struct)				\
 	size_t _struct##_size_rh;					\
 	RH_KABI_EXCLUDE(struct _struct##_rh *_struct##_rh)
+#define RH_KABI_SIZE_AND_EXTEND_PTR(_struct)				\
+	_RH_KABI_SIZE_AND_EXTEND_PTR(_struct)

-#define RH_KABI_SIZE_AND_EXTEND(_struct)				\
+#define _RH_KABI_SIZE_AND_EXTEND(_struct)				\
 	size_t _struct##_size_rh;					\
 	RH_KABI_EXCLUDE(struct _struct##_rh _struct##_rh)
+#define RH_KABI_SIZE_AND_EXTEND(_struct)				\
+	_RH_KABI_SIZE_AND_EXTEND(_struct)

 /*
  * RH_KABI_SET_SIZE calculates and sets the size of the extended struct and
-- 
2.26.2

